pkg/reconciler/managed/reconciler.go

```diff
@@ -693,17 +726,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	// not realize that the controller is still trying to reconcile
 	// (and modify or delete) the resource since they forgot to enable the
 	// feature flag.
-	if !r.managementPoliciesEnabled && (managed.GetManagementPolicy() == xpv1.ManagementObserveOnly || managed.GetManagementPolicy() == xpv1.ManagementOrphanOnDelete) {
-		log.Debug(errManagementPolicy, "policy", managed.GetManagementPolicy())
-		record.Event(managed, event.Warning(reasonManagementPolicyNotEnabled, errors.New(errManagementPolicy)))
-		managed.SetConditions(xpv1.ReconcileError(errors.New(errManagementPolicy)))
+	if isNonDefaultManagementPolicy(managementPoliciesEnabled, managed.GetManagementPolicy()) {
+		log.Debug(errManagementPolicyNonDefault, "policy", managed.GetManagementPolicy())
+		record.Event(managed, event.Warning(reasonManagementPolicyNotEnabled, errors.New(errManagementPolicyNonDefault)))
+		managed.SetConditions(xpv1.ReconcileError(errors.New(errManagementPolicyNonDefault)))
+		return reconcile.Result{}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
+	}
```

Comment:
I wonder whether there's any value in debug logging the management policy even in the happy path? I could imagine adding it to all logs using `log.WithValues`.

Reply:
Added it to log.WithValues, but only if the management policies feature is enabled. Makes sense to add it to the logs in my opinion.